### PR TITLE
Make queue capacity configurable

### PR DESCRIPTION
**Pull request by rightfold**

### CODE REVIEW

File: server/src/Config.hs [hunk 1]
```diff
@@ -59,6 +61,11 @@ configParser environment = Config
                   metavar "HOST:PORT" <>
                   help "If provided, Icepeak collects various metrics and provides them to Prometheus on the given endpoint."
                  ))
+  <*> fmap (fromMaybe 256)
```
Comment (fatho) [inline server/src/Config.hs]: It's possible to add a default value with the `value` modifier, i.e.

```
(option auto
                       (long "queue-capacity" <>
                        metavar "INTEGER" <>
                        value 256 <>
                        help "Higher values allow more requests to come in within a short amount of time.")
```

looks a bit cleaner then.

Comment (rightfold) [reply server/src/Config.hs]: Ah, nice. :+1:

File: server/src/Config.hs [hunk 2]
```diff
@@ -59,6 +61,11 @@ configParser environment = Config
...
+           (optional (option auto
+                       (long "queue-capacity" <>
+                        metavar "INTEGER" <>
+                        help "Higher values allow more requests to come in within a short amount of time.")))
```
Comment (ruuda) [inline server/src/Config.hs]: Maybe mention that this is a trade-off between the amount of data risked losing, the ability to accept bursts of requests without denying them.

Comment (rightfold) [reply server/src/Config.hs]: Good point. :+1:

File: server/src/Config.hs [hunk 3]
```diff
@@ -59,6 +60,12 @@ configParser environment = Config
                   metavar "HOST:PORT" <>
                   help "If provided, Icepeak collects various metrics and provides them to Prometheus on the given endpoint."
                  ))
+  <*> option auto
+       (long "queue-capacity" <>
+        metavar "INTEGER" <>
+        value 256 <>
+        help ("Higher values allow more requests to come in within a short amount of time. " <>
+              "Smaller values increase the risk of data loss."))
```
Comment (rkrzr) [inline server/src/Config.hs]: s/increase/decrease

If the queue is smaller, fewer values can be lost.

Comment (rightfold) [reply server/src/Config.hs]: Right, but you also lose more requests. Will clarify and merge.

**Pull request merged**